llvm/tools/sycl-post-link/ModuleSplitter.h

```diff
@@ -34,66 +35,122 @@ enum IRSplitMode {
 };
 
 // A vector that contains all entry point functions in a split module.
-using EntryPointVec = std::vector<const Function *>;
+using EntryPointSet = SetVector<const Function *>;
+
+enum class SyclEsimdSplitStatus { SYCL_ONLY, ESIMD_ONLY, SYCL_AND_ESIMD };
+
+// Describes scope covered by each entry in the module-entry points map
+// populated by the groupEntryPointsByScope function.
+enum EntryPointsGroupScope {
+  Scope_PerKernel, // one entry per kernel
+  Scope_PerModule, // one entry per module
+  Scope_Global     // single entry in the map for all kernels
+};
 
 // Represents a named group of device code entry points - kernels and
 // SYCL_EXTERNAL functions. There are special group names - "<SYCL>" and
 // "<ESIMD>" - which have effect on group processing.
-
 struct EntryPointGroup {
-  StringRef GroupId;
+  // Properties an entry point (EP) group
+  struct Properties {
+    // Whether all EPs are ESIMD, SYCL or there are both kinds.
+    SyclEsimdSplitStatus HasESIMD = SyclEsimdSplitStatus::SYCL_AND_ESIMD;
+    // Whether any of the EPs use double GRF mode.
+    bool UsesDoubleGRF = false;
+    // Scope represented by EPs in a group
+    EntryPointsGroupScope Scope = Scope_Global;
+
+    Properties merge(const Properties &Other) const {
+      Properties Res;
+      Res.HasESIMD = HasESIMD == Other.HasESIMD
+                         ? HasESIMD
+                         : SyclEsimdSplitStatus::SYCL_AND_ESIMD;
+      Res.UsesDoubleGRF = UsesDoubleGRF || Other.UsesDoubleGRF;
```

Comment:
Based on this, I suppose using double GRF is preferred to not using double GRF?

Reply:
double GRF  is used by request  that need to be honored. Thus that property remains during the merge.

Reply:
Got it. Thanks.